### PR TITLE
Refactor Presets and Supporting Documentation

### PR DESCRIPTION
**Pull request by indiegenus**
# Notes

- updated all Presets and their respective pages in Docs
- cleaned up and reorganized the sidebar menu for Docs
  - removed "Existing Project" page
  - removed Presets "Introduction" page
  - moved "Installation" and "How It Works" pages up to top-level with "Introduction" and "Demo"
  - renamed "Adonis" preset page to "AdonisJS"
  - renamed "Wordpress" preset page to "WordPress" (with uppercase "P")
  - added a new section called "Snippets", with first snippet for adding "MailHog" to a local dev environment
- WordPress preset:
  - added a `.env` file to the project to make `kool run mysql` work
  - changed DB_DATABASE to match WP default `wordpress` (instead of `database`); via the new `.env` file
- Hugo preset:
  - added a `quickstart` script in the `kool.yml` to facilitate the creation of a Hello World experience
- added new basic, Hello World presets for PHP and NodeJS
  - for now, at least, I decided to keep the "Starting New Project" page as is, and **not** use the new PHP preset, because it better explains how to add Kool to **any** project, when a preset is not available.

# TODO

- [ ] fix permissions issues with Hugo and WordPress (see https://github.com/kool-dev/kool/issues/316)
- [ ] fix the absence of commented out lines in the auto-generated `kool.yml` files (see https://github.com/kool-dev/kool/issues/312#issuecomment-830670212)
- [ ] move all Presets out of 2-Presets into 1-Getting Started and deprecate 2-Presets as a section in the sidebar menu
- [ ] finish testing all variants (based on wizard selections)

| Issue | # |
| -----: | :-----: |
| :pencil: Refactor | Yes |
| :open_book: Documentation | Yes |

### CODE REVIEW

File: docs/2-Presets/WordPress.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+? What cache service do you want to use [Use arrows to move, type to filter]
+> Redis 6.0
+  Memcached 1.6
+  none
+
+? What javascript package manager do you want to use [Use arrows to move, type to filter]
+> npm
+  yarn
+
+? What composer version do you want to use [Use arrows to move, type to filter]
```
Comment (fabriciojs) [inline docs/2-Presets/WordPress.md:49]: Indeed we seemed to already have Composer inquiry for creating Wordpress app... does that even make sense? Should we remove?

File: docs/5-Snippets/Mailhog.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+# Mailhog
```
Comment (fabriciojs) [inline docs/5-Snippets/Mailhog.md:1]: Just a comment - we have very common snippets for `elasticsearch`, `ssh tunneling`, `caddy proxy` - which we can add next to have a better body of content in this section to start with.

**Review by fabriciojs — APPROVED** [review]
Great job @indiegenus 👍🏻

File: docs/2-Presets/WordPress.md (hunk 1 above)
Comment (indiegenus) [reply docs/2-Presets/WordPress.md:49]: @fabriciojs - I think you're right, that it's not applicable to WP and we should remove. Awaiting agreement from @dbpolito

Comment (dbpolito) [reply docs/2-Presets/WordPress.md:49]: yup, we can remove composer from wordpress...

**Review by dbpolito — APPROVED** [review]

**Review by danielsuguimoto — APPROVED** [review]

**Comment by indiegenus** [comment]
@dbpolito @fabriciojs - see https://github.com/kool-dev/kool/pull/317/commits/d86b005fb951625bd13f96a1f59646f25d90489f with updates to PR based on your feedback:
- removed Composer from WordPress preset
- added permissions-related comments to Hugo and WordPress docs (for Linux and Windows WSL devs)

**Pull request merged**